packages/hdwallet-ledger-webusb/src/transport.ts

```diff
@@ -86,18 +95,76 @@ export class LedgerWebUsbTransport extends LedgerTransport {
   }
 
   public async call(coin: string, method: string, ...args: any[]): Promise<LedgerResponse> {
+    return  await this.sendToLedger({
+      fromCallFn: true,
+      args,
+      method,
+      coin
+    })
+  }
+
+  public async sendToLedger(sendArgs: LedgerRequest): Promise<LedgerResponse> {
+    let { action, coin, method, args, fromCallFn } = sendArgs
     let response
 
-    try {
-      this.emit(`ledger.${coin}.${method}.call`, makeEvent({
-        message_type: method,
-        from_wallet: false,
-        message: {}
-      }))
+    // set it back to false in case it was true
+    if (this.isCancelled) this.isCancelled = false
 
-      response = await new (translateCoin(coin))(this.transport)[method](...args)
+    try {
+      await retry(async (bail) => {
+        if (this.isCancelled) {
+          this.isCancelled = false
+          bail()
+          return
+        }
+
+        // open transport if it's closed
+        await this.open()
```

Comment:
Is `open()` idempotent? what happens for `open(); open();`?

Reply:
@keepkeyjon Yes as far as I can tell. When open is called multiple times, it opens the current transport and does not cause side effects if it is already open.